src/test/java/test/io/ticktok/client/tick/HttpTickerPolicyTest.java

```diff
@@ -0,0 +1,185 @@
+        server = new TicktokServer().onPopFailWith(500);
+        assertNoTicksWereConsumed();
+        assertTrue(server.popped);
+    }
+
+    @Test
+    void invokeOnTick() throws Exception {
+        server = new TicktokServer().tickOn("tick-me");
+        PacedTickCounter tickCounter = new PacedTickCounter(2);
+        register("tick-me", tickCounter::countDown);
+        tickCounter.assertNoTicks();
+        assertTimeoutPreemptively(ofSeconds(2), tickCounter::await, "No ticks");
+    }
+
+    @Test
+    void ignoreWhenThereNoTicksToPop() throws Exception {
+        server = new TicktokServer();
+        assertNoTicksWereConsumed();
+    }
+
+    @Test
+    void disconnectAllClocks() throws Exception {
+        server = new TicktokServer();
+        AtomicInteger tickCount = new AtomicInteger();
+        register("c1", tickCount::incrementAndGet);
+        register("c2", tickCount::incrementAndGet);
+        tickListener.disconnect();
+        sleep(100);
+        server.tickOn("c1", "c2");
+        assertNoTicks(tickCount);
+    }
+
+    @Test
+    void allowToRegisterAfterDisconnect() throws Exception {
+        server = new TicktokServer().tickOn("c1");
+        tickListener.disconnect();
+        CountDownLatch tickConsumed = new CountDownLatch(1);
+        register("c1", tickConsumed::countDown);
+        assertTimeoutPreemptively(ofSeconds(1), (Executable) tickConsumed::await);
+    }
+
+    @Test
+    void replaceCallbackForAGivenClock() throws Exception {
+        server = new TicktokServer();
+        CountDownLatch countForOriginal = new CountDownLatch(1);
+        CountDownLatch countForNew = new CountDownLatch(1);
+        register("c1", countForOriginal::countDown);
+        register("c1", countForNew::countDown);
+        server.tickOn("c1");
+        assertTimeoutPreemptively(ofSeconds(1), (Executable) countForNew::await);
+        assertThat("original consumer should not get ticks", countForOriginal.getCount(), is(1L));
+    }
+
+    @AfterEach
+    void stopServer() {
+        tickListener.disconnect();
+        if (server != null) {
+            server.stop();
+        }
+    }
+
+    static class TicktokServer extends BlinkServer {
+
+        public static final int PORT = 9191;
+
+        public boolean popped;
+        private int statusCode = 200;
+        private Set<String> clocks = new HashSet<>();
+
+        public TicktokServer() throws IOException {
+            super(PORT);
+            get("/{id}/pop", (req, res) -> {
+                popped = true;
+                res.status(statusCode);
+                if (clocks.contains(req.pathParam("id"))) {
+                    return asList(new JsonObject());
+                }
+                return new ArrayList<>();
+            });
+        }
+
+        public TicktokServer tickOn(String... ids) {
+            this.clocks.addAll(asList(ids));
+            return this;
+        }
+
+        public TicktokServer onPopFailWith(int code) {
```

Comment:
name is not clear. you state that it indicate fail, but you only save a status...

Reply:
this is a helper function and it should be clear to the user which is more important then how it's doing it 
I think new TicktokServer().onPopFailWith(404) is clear and more important then its implementation

Reply:
i think otherwise, implementation should match your statement.
its not good practice to state an action your implementation is not supporting.
i would do for example (in the current implementation you choose) the following:

`new TicktokServer().markFailStatus(404).onPop()`.

404 status is a well known in software. its actually self explanatory

Reply:
I can change it to whenPopFailWith(404) 
which will be like mockito

Reply:
ok, that's an option

Reply:
will do